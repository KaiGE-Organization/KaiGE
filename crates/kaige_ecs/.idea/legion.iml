<?xml version="1.0" encoding="UTF-8"?>
<module type="JAVA_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/examples" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/tests" isTestSource="true" />
      <sourceFolder url="file://$MODULE_DIR$/benches" isTestSource="true" />
      <sourceFolder url="file://$MODULE_DIR$/legion_systems/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/legion_systems/examples" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/legion_systems/tests" isTestSource="true" />
      <sourceFolder url="file://$MODULE_DIR$/legion_systems/benches" isTestSource="true" />
      <sourceFolder url="file://$MODULE_DIR$/legion_core/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/legion_core/examples" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/legion_core/tests" isTestSource="true" />
      <sourceFolder url="file://$MODULE_DIR$/legion_core/benches" isTestSource="true" />
      <excludeFolder url="file://$MODULE_DIR$/legion_core/target" />
      <excludeFolder url="file://$MODULE_DIR$/legion_systems/target" />
      <excludeFolder url="file://$MODULE_DIR$/target" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>