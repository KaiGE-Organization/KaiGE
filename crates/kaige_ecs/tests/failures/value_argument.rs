use legion::system;

#[system]
fn value_arguement(_: usize) {}

fn main() {}
