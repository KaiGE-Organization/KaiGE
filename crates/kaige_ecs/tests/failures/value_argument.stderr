error: system function parameters must be `CommandBuffer` or `SubWorld` references, [optioned] component references, state references, or resource references
 --> $DIR/value_argument.rs:4:23
  |
4 | fn value_arguement(_: usize) {}
  |                       ^^^^^
